### PR TITLE
Bias Monitor/webpage for all instruments

### PR DESCRIPTION
**Pull request by bsunnquist**
This PR incorporates all relevant instruments (NIRCam, NIRISS, NIRSpec) into the Bias Monitor/database/webpage, as well as some other enhancements/fixes after talking with the instrument reps. Changes include:

- added the histogram distribution of the calibrated 0th group image to the database tables, and displayed this for the NIRISS webpage
- added a test for this histogram calculation to `test_bias_monitor.py`
- flipped the NIRSpec odd/even calculation after talking with Maria, which also fixes the NIRSpec amp boundary issue (https://github.com/spacetelescope/jwql/issues/718)
- confirmed the changes by @penaguerrero are still taken into account in this PR (i.e. the Bias portion of https://github.com/spacetelescope/jwql/issues/600 is completed)
- mimicked the Readnoise Monitor's image display strategy (which results in a clearer image on the webpage), most recent search function (fixes a bug), and new image search/output file strategy (which will make it easier if we ever need to reprocess everything)
- changed the plot scale limits to better deal with outliers
- switched to using our generalized pipeline function from `pipeline_tools.py` for the image calibration

### CODE REVIEW

**Comment by bsunnquist** [comment]
@bourque this is all ready for review whenever

**Comment by bourque** [comment]
Thank you @bsunnquist!  I am going to try to find someone to review this.  In the meantime, it looks like the GitHub Actions build is failing due to failing tests.  Could you see what is going on there?

**Comment by bsunnquist** [comment]
@bourque sounds good - the testing issue is fixed now too

File: jwql/tests/test_bias_monitor.py [hunk 1]
```diff
@@ -102,3 +103,21 @@ def test_identify_tables():
...
+    data = np.zeros((100, 100))
+    counts_truth = [10000]
+    bin_centers_truth = [0.0]
+
+    # Find the histogram stats of the test data using the bias monitor
+    counts, bin_centers = monitor.make_histogram(data)
+    counts, bin_centers = list(counts), list(bin_centers)
+
+    assert counts == counts_truth
+    assert bin_centers == bin_centers_truth
```
Comment (bourque) [inline jwql/tests/test_bias_monitor.py:123]: Great work on this, and thank you for expanding the test suite!

**Review by bourque — APPROVED** [review]
Thank you for all of these fixes/improvements @bsunnquist! Excellent work!  Everything looks good to me, and works well when I test the relevant code and display the webpages on my end.

**Pull request merged**